flocker/control/test/test_httpapi.py

```diff
@@ -1309,6 +1309,143 @@ def test_container_with_volume(self):
+        ))
+
+        def updated(_):
+            deployment = self.persistence_service.get()
+            expected = Deployment(
+                nodes={
+                    Node(
+                        hostname=self.NODE_A,
+                        applications=[
+                            Application(
+                                name='mycontainer',
+                                image=DockerImage.from_string('busybox')
+                            ),
+                        ]
+                    ),
+                    Node(hostname=self.NODE_B),
+                }
+            )
+            self.assertEqual(deployment, expected)
+
+        d.addCallback(updated)
+        return d
+
+    def test_update_new_host(self):
+        """
+        An API request to update a named container's host to a different host
+        results in an updated configuration.
+        """
+        d = self._create_container()
+
+        d.addCallback(lambda _: self.assertResponseCode(
+            b"POST", b"/configuration/containers/mycontainer",
+            {u"host": self.NODE_B}, OK
+        ))
+
+        def updated(_):
+            deployment = self.persistence_service.get()
+            expected = Deployment(
+                nodes={
+                    Node(hostname=self.NODE_A),
+                    Node(
+                        hostname=self.NODE_B,
+                        applications=[
+                            Application(
+                                name='mycontainer',
+                                image=DockerImage.from_string('busybox')
+                            ),
+                        ]
+                    ),
+                }
+            )
+            self.assertEqual(deployment, expected)
+
+        d.addCallback(updated)
+        return d
+
+    def test_update_invalid_container_name(self):
+        """
+        An API request to update a named container's host to a different host
+        results in an error if the named container does not exist.
+        """
+        return self.assertResult(
+            b"POST", b"/configuration/containers/somecontainer",
+            {u"host": self.NODE_A}, NOT_FOUND,
+            {u"description": u"Container not found."},
+        )
+
+    def test_response(self):
+        """
+        An API request to move a container to a new host returns the
+        expected JSON response.
+        """
+        d = self._create_container()
+
+        def update_container(_):
+            request = {u"host": self.NODE_B}
+            result = {
+                u"host": self.NODE_B,
+                u"name": u"mycontainer",
+                u"image": u"busybox:latest",
+                u"restart_policy": {u"name": u"never"}
+            }
+            return self.assertResult(
+                b"POST", b"/configuration/containers/mycontainer",
+                request, OK, result
+            )
+
+        d.addCallback(update_container)
+        return d
+
```

Comment:
You didn't implement or test the bit where datasets are moved along with the application.